velox/functions/lib/Re2Functions.cpp

```diff
@@ -404,6 +404,19 @@ bool matchSuffixPattern(
           length) == 0;
 }
 
+// Search for 'pattern' in 'input' starting from positions greater than
+// 'startPosition'.
+vector_size_t fixedPatternSearch(
+    const std::string input,
+    const std::string pattern,
+    vector_size_t startPosition) {
+  vector_size_t patternIdx = input.find(pattern, startPosition);
```

Comment:
Do we need to use 'find' here ? memcmp(input.data() + startPosition, ...) should work I think. That would let you change the input parameters to StringView also.

Reply:
Thanks for the feedback, I figured find would be better instead of doing memcmp recursively till the fixed pattern is (not) found, should I use a helper find function which uses memcmp internally?

Reply:
Yeah that will be better I think. The strings could be very big here. Its better to use StringView.